sources/api/pluto/src/main.rs

```diff
@@ -332,55 +319,42 @@ async fn get_private_dns_name(client: &mut ImdsClient) -> Result<String> {
         .context(error::Ec2Snafu)
 }
 
-/// Print usage message.
-fn usage() -> ! {
-    let program_name = env::args().next().unwrap_or_else(|| "program".to_string());
-    eprintln!(
-        r"Usage: {} [max-pods | cluster-dns-ip | node-ip | provider-id | private-dns-name]",
-        program_name
-    );
-    process::exit(1);
-}
-
-/// Parses args for the setting key name.
-fn parse_args(mut args: env::Args) -> String {
-    args.nth(1).unwrap_or_else(|| usage())
-}
-
 async fn run() -> Result<()> {
-    let setting_name = parse_args(env::args());
     let mut client = ImdsClient::new();
 
-    let setting = match setting_name.as_ref() {
-        "cluster-dns-ip" => get_cluster_dns_ip(&mut client).await,
-        "node-ip" => get_node_ip(&mut client).await,
-        // If we want to specify a reasonable default in a template, we can exit 2 to tell
-        // sundog to skip this setting.
-        "max-pods" => get_max_pods(&mut client)
-            .await
-            .map_err(|_| process::exit(2)),
-        "provider-id" => get_provider_id(&mut client).await,
-        "private-dns-name" => get_private_dns_name(&mut client).await,
-        _ => usage(),
-    }?;
-
-    // sundog expects JSON-serialized output so that many types can be represented, allowing the
-    // API model to use more accurate types.
-
-    // 'max_pods' setting is an unsigned integer, convert 'settings' to u32 before serializing to JSON
-    if setting_name == "max-pods" {
-        let max_pods = serde_json::to_string(
-            &setting
-                .parse::<u32>()
-                .context(error::ParseToU32Snafu { setting: &setting })?,
-        )
-        .context(error::OutputJsonSnafu { output: &setting })?;
-        println!("{}", max_pods);
+    // Generate the node-ip
+    let node_ip = get_node_ip(&mut client).await?;
+    // Generate the cluster-dns-ip
+    let cluster_dns_ip = get_cluster_dns_ip(&mut client).await?;
+
+    // Generate max-pods
+    // Skip the set if this fails
+    let max_pods = if let Ok(max_pods) = get_max_pods(&mut client).await {
+        Some(max_pods)
```

Comment:
Why skip the set? This seems to be new behavior - before we would error out if we couldn't get a max pods value. The conditional was just to deal with sundog's weird output requirements; we could print the number directly but needed to JSON-format the other values.

Reply:
Discussed offline, we should be good here.